realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1865,3 +1865,40 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeIsEmpty
         }
     } CATCH_STD()
 }
+
+JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeDistinct
+    (JNIEnv *env, jobject, jlong nativeQueryPtr, jlongArray columnIndexes) {
+
+    JniLongArray arr(env, columnIndexes);
+    jlong column_idx = arr[arr.len() - 1];
+    TableRef table_ref = getTableForLinkQuery(nativeQueryPtr, arr);
```

Comment:
No, `distinct()` takes a vector as argument, and it is the combination of columns which will have unique values.